### DISCUSSION TITLE
Shared IRQs on STM32 ADC

### DISCUSSION THREAD

**Discussion by davedesro** [zephyrproject-rtos/zephyr#63821] [RFC]
There's a well known and annoying issue around zephyr and shared irqs.  I fully appreciate how complex a proper solution is, and the time it takes to get it right.

There is one small part, though, with stm32's existing work-around with shared irqs in the adc block. 
Here's [Good recent discussion](https://github.com/zephyrproject-rtos/zephyr/discussions/61263) on the topic. 
The existing ADC_STM32_STHARED_IRQS code is quite limiting. **You can either use adc blocks that have the same IRQn, or adc blocks that have unique IRQns.. but not both!**

I'm wondering.. as an interim solution, can we make the ADC code a bit more robust? 

I hope to provide a patch for comments in the near future.

**Comment by gautierg-st** [comment]
Hello,
PR #61422 (about shared interrupt support) has been merged since the discussion you mentioned. I haven't had time to look into it, but it should considerably ease the process of supporting all ADC IRQ, and I'd be happy to review if you want to tackle it.

**Reply by davedesro** [reply]
Thank you @gautierg-st for the notification 👍
